platform/security/platform-security-core-api/src/main/java/ddf/security/principal/GuestPrincipal.java

```diff
@@ -65,9 +68,25 @@ public static String parseAddressFromName(String fullName) {
         if (!StringUtils.isEmpty(fullName)) {
             String[] parts = fullName.split(NAME_DELIMITER);
             if (parts.length == 2) {
-                return parts[1];
+                return formatIpAddress(parts[1]);
             }
         }
         return null;
     }
+
+    //IPv6 addresses should be contained within brackets to conform
+    //to the spec IETF RFC 2732
+    private static String formatIpAddress(String ipAddress) {
+        try {
+            if (InetAddress.getByName(ipAddress) instanceof Inet6Address) {
```

Comment:
I tried it out but those methods are not available in the version of StringUtils utilized in this module.